packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -97,9 +97,11 @@ export function RepositoryMixin<T extends Class<any>>(superClass: T) {
      * }
      * ```
      */
-    dataSource(dataSource: juggler.DataSource, name?: string) {
+    dataSource(dataSource: Class<juggler.DataSource>, name?: string) {
       const dataSourceKey = `datasources.${name || dataSource.name}`;
-      this.bind(dataSourceKey).to(dataSource);
+      this.bind(dataSourceKey)
+        .toClass(dataSource)
+        .tag('datasource');
```

Comment:
As mentioned in another comment, we must change the scope of all datasource binding to a SINGLETON.